tgui/packages/tgui/interfaces/DecalPainter.tsx

```diff
@@ -0,0 +1,131 @@
+import { useBackend } from '../backend';
+import { Button, ColorBox, Flex, Section } from '../components';
+import { Window } from '../layouts';
+
+type Decal = {
+  name: string,
+  decal: string
+}
+
+type Color = {
+  name: string,
+  color: string
+}
+
+type Dir = {
+  name: string,
+  dir: number
+}
+
+type DecalPainterData = {
+  icon_prefix: string,
+  decal_list: Decal[],
+  color_list: Color[],
+  dir_list: Dir[],
+  nondirectional_decals: string[],
+  current_decal: string,
+  current_color: string,
+  current_dir: number
+}
+
+export const DecalPainter = (props, context) => {
+  const { act, data } = useBackend<DecalPainterData>(context);
+
+  return (
+    <Window
+      width={550}
+      height={400}>
+      <Window.Content>
+        <Section title="Decal Color">
+          {data.color_list.map(color => {
+            return (
+              <Button
+                key={color.color}
+                selected={color.color === data.current_color}
+                onClick={() => act('select color', {
+                  color: color.color,
+                })}
+              >
+                <ColorBox color={color.color} mr={0.5} />
+                { color.name }
+              </Button>
+            );
+          })}
+        </Section>
+        <Section title="Decal Style">
+          <Flex
+            direction="row" wrap="nowrap"
+            align="fill" justify="fill"
+          >
+            {data.decal_list.map(decal => {
+              const nondirectional = data.nondirectional_decals
+                .includes(decal.decal);
+
+              return nondirectional ? (
+                // Tallll button for nondirectional
+                <IconButton
+                  decal={decal.decal} dir={2}
+                  label={decal.name}
+                  selected={decal.decal === data.current_decal}
+                />
+              ) : (
+                // 4 buttons for directional
+                <Flex
+                  key="decal.decal"
```

Comment:
I suppose this key is static because there's only one item in the list, but there should be some guarantees that this is a unique string in the list, because if there was more than one of this item with key "decal.decal", things would break horribly.

Reply:
Fixed